### PR TITLE
Set PNN instead of PSNs for wmbs file locations

### PR DESCRIPTION
**Pull request by amaltaro**
T0 version of https://github.com/dmwm/WMCore/pull/8280

@hufnagel @ticoann please review. Dirk, can you ask T0 Ops to run a replay to test these changes (once you're happy with them)?

### CODE REVIEW

**Comment by hufnagel** [comment]
What WMCore tag should this use ? T0 Ops can't run a replay until I at least confirm basic compatibility with that tag. Currently T0 is at 1.1.6. Guess I should switch directly yo 1.1.10 ?

**Comment by ticoann** [comment]
The tag which include this change is 1.1.10.pre1

File: src/python/T0/WMBS/Oracle/RunConfig/InsertStreamer.py [hunk 1]
```diff
@@ -29,8 +29,8 @@ def execute(self, binds, conn = None, transaction = False):
                          (FILEID, RUN, LUMI)
                          VALUES (wmbs_file_details_SEQ.nextval, :RUN, :LUMI)
                        INTO wmbs_file_location
-                         (FILEID, LOCATION)
-                         VALUES (wmbs_file_details_SEQ.nextval, (SELECT id FROM wmbs_location WHERE site_name = 'T0_CH_CERN'))
+                         (FILEID, PNN)
+                         VALUES (wmbs_file_details_SEQ.nextval, (SELECT id FROM wmbs_pnns WHERE pnn = 'T0_CH_CERN'))
```
Comment (hufnagel) [inline src/python/T0/WMBS/Oracle/RunConfig/InsertStreamer.py]: WHERE pnn = 'T0_CH_CERN_Disk'

**Comment by hufnagel** [comment]
Replay still dies with

2017-12-08 18:51:37,053:140257366316800:ERROR:BaseWorkerThread:Error in worker algorithm (1):
Backtrace:
  <T0Component.Tier0Feeder.Tier0FeederPoller.Tier0FeederPoller object at 0x7f9036806410> 'NoneType' object is not iterable  File "/data/tier0/srv/wmagent/2.1.0/sw/slc6_amd64_gcc493/cms/wmagent/1.1.6.patch5/lib/python2.7/site-packages/WMCore/WorkerThreads/BaseWorkerThread.py", line 183, in __call__
    tSpent, results, _ = self.algorithm(parameters)

2017-12-08 18:51:37,053:140257366316800:INFO:Harness:>>>Terminating worker threads
2017-12-08 18:51:37,068:140257366316800:ERROR:BaseWorkerThread:Error in event loop (2): <T0Component.Tier0Feeder.Tier0FeederPoller.Tier0FeederPoller object at 0x7f9036806410> 'NoneType' object is not iterable
Backtrace:
  File "/data/tier0/srv/wmagent/2.1.0/sw/slc6_amd64_gcc493/cms/wmagent/1.1.6.patch5/lib/python2.7/site-packages/WMCore/WorkerThreads/BaseWorkerThread.py", line 211, in __call__
    raise ex

2017-12-08 18:51:37,068:140257366316800:INFO:BaseWorkerThread:Worker thread <T0Component.Tier0Feeder.Tier0FeederPoller.Tier0FeederPoller object at 0x7f9036806410> terminated

Have you run any functional tests with 1.1.10.pre1 yet ?

File: src/python/T0/WMBS/Oracle/RunConfig/InsertStreamer.py (hunk 1 above)
Comment (amaltaro) [reply src/python/T0/WMBS/Oracle/RunConfig/InsertStreamer.py]: I'll update it soon.

**Comment by amaltaro** [comment]
About this failure on Tier0FeederPoller. You need to update the code of any components (inheriting from `BaseWorkerThread`) with a decorator for the `algorithm` method
https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/ErrorHandler/ErrorHandlerPoller.py#L29
https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/ErrorHandler/ErrorHandlerPoller.py#L370

See original PR changes here https://github.com/dmwm/WMCore/pull/8165

**Comment by hufnagel** [comment]
Seems to be working now

2017-12-08 21:14:15,512:140208127547136:INFO:BaseWorkerThread:Tier0FeederPoller took 65.127 secs to execute

Will keep watching the replay to make sure data discovery in the JobCreator also works (since these queries were also changed).

**Comment by hufnagel** [comment]
You forgot to change a GROUP statement in GetAvailableRepackMergeFiles.py, JobCreator crashes in the RepackMerge subscriptions.

**Comment by amaltaro** [comment]
Sorry Dirk, GetAvailableRepackMergeFiles.py has been fixed.

**Comment by hufnagel** [comment]
Ok, tested in a replay, works now. Merging this.

**Comment by amaltaro** [comment]
thanks!

**Pull request merged**